### PR TITLE
config, variable: add a config to set 'lower_case_table_names' system variable.

### PR DESCRIPTION
**Pull request by winkyao**
Add a config item `lower-case-table-names ` to modify system variable `lower_case_table_names`.

ref https://dev.mysql.com/doc/refman/5.7/en/identifier-case-sensitivity.html

### CODE REVIEW

**Comment by coocood** [comment]
Need to validate the input value to be 0, 1 or 2?

**Comment by shenli** [comment]
LGTM
Please fix the CI.

**Comment by winkyao** [comment]
@coocood PTAL

File: config/config.go [hunk 1]
```diff
@@ -52,6 +52,9 @@ type Config struct {
 	TokenLimit      uint   `toml:"token-limit" json:"token-limit"`
 	OOMAction       string `toml:"oom-action" json:"oom-action"`
 	EnableStreaming bool   `toml:"enable-streaming" json:"enable-streaming"`
+	// Set sys variable lower-case-table-names, ref: https://dev.mysql.com/doc/refman/5.7/en/identifier-case-sensitivity.html.
+	// TODO: we now just ignore the system variable lower_case_table_names, we treat the table name as case sensitive actually.
```
Comment (coocood) [inline config/config.go]: We actually only support mode 2, which keeps the original case, but the comparison is case-insensitive.

File: config/config.toml.example [hunk 2]
```diff
@@ -41,6 +41,9 @@ oom-action = "log"
 # Enable coprocessor streaming.
 enable-streaming = false
 
+# Set system variable 'lower_case_table_names'
```
Comment (shenli) [inline config/config.toml.example]: Add a comment here for the usage. Could we do not put it here and use the default value in config.go?

Comment (winkyao) [reply config/config.toml.example]: The config_test.go will check the value in config.toml.example is the same as cfg.LowerCaseTableNames, so we need to put an example here.

**Comment by coocood** [comment]
LGTM

File: config/config.toml.example (hunk 2 above)
Comment (shenli) [reply config/config.toml.example]: Got it!

**Pull request merged**